File: _subsets.ts
function subsets(nums: number[]): number[][] {

    let res =[]
    let subset = []

    let dfs =(i, subset) =>{
        if(i==nums.length){
            res.push(subset)
            return
        }

        dfs(i+1, [...subset, nums[i]])

        dfs(i+1, subset)
    }

    dfs(0,subset)

    return res


};

File: Python/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:

        global res

        res = []

        def util(i, cur):
            if(i>=len(nums)):
                res.append(cur.copy())
                return 
                
            cur.append(nums[i])
            util(i+1, cur)
            cur.pop()
            util(i+1,cur)
        
        util(0,[])
        
        return res
